zabbix/stackstate_checks/zabbix/zabbix.py

```diff
@@ -0,0 +1,366 @@
+        return "ZabbixHostGroup(host_group_id:%s, name:%s.)" % (self.host_group_id, self.name)
+
+
+class ZabbixTrigger:
+    def __init__(self, trigger_id, description, priority):
+        self.trigger_id = trigger_id
+        self.description = description
+        self.priority = priority  # translates to severity
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return "ZabbixTrigger(trigger_id:%s, description:%s, priority:%s.)" % (self.trigger_id, self.description,
+                                                                               self.priority)
+
+
+class ZabbixEvent:
+    def __init__(self, event_id, acknowledged, host_ids, trigger):
+        self.event_id = event_id
+        self.acknowledged = acknowledged  # 0/1 (not) acknowledged
+        self.host_ids = host_ids
+        self.trigger = trigger  # ZabbixTrigger
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return "ZabbixEvent(event_id:%s, acknowledged:%s, host_ids:%s, trigger:%s)" % (self.event_id, self.acknowledged,
+                                                                                       self.host_ids, self.trigger)
+
+
+class ZabbixProblem:
+    def __init__(self, event_id, acknowledged, trigger_id, severity):
+        self.event_id = event_id
+        self.acknowledged = acknowledged
+        self.trigger_id = trigger_id
+        self.severity = severity
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return "ZabbixProblem(event_id:%s, acknowledged:%s, trigger_id:%s, severity:%s)" % \
+               (self.event_id, self.acknowledged, self.trigger_id, self.severity)
+
+
+class ZabbixCheck(AgentCheck):
+    SERVICE_CHECK_NAME = INSTANCE_TYPE = "Zabbix"
+    begin_epoch = None  # start to listen to events from epoch timestamp
+
+    def __init__(self, name, init_config, instances=None):
+        AgentCheck.__init__(self, name, init_config, instances)
+        self.url = None
+        self.user = None
+        self.password = None
+
+    def get_instance_key(self, instance):
+        if 'url' not in instance:
+            raise ConfigurationError('Missing API url in configuration.')
+
+        return TopologyInstance(self.INSTANCE_TYPE, instance["url"])
+
+    def check(self, instance):
+        """
+        Integration logic
+        """
+        if 'user' not in instance:
+            raise ConfigurationError('Missing API user in configuration.')
+        if 'password' not in instance:
+            raise ConfigurationError('Missing API password in configuration.')
+
+        stackstate_environment = instance.get('stackstate_environment', 'Production')
+        self.ssl_verify = instance.get('ssl_verify', True)
+
+        url = instance['url']
+
+        topology_instance = {
+            "type": self.SERVICE_CHECK_NAME,
+            "url": url
+        }
+        try:
+            self.check_connection(url)
```

Comment:
can the `check_connection` or `login` fail?
if it does the `stop_snapshot` is called without a `start_snapshot` first, that may cause problems
also can we test this ?

Reply:
moved `start_snapshot` to first line in `try` block to avoid the failure.